### ISSUE TITLE
Confusing error when having declarations before the toplevel module

### ISSUE THREAD

**Issue by GoogleCodeExporter** [agda/agda#1077]
This of on the checker of  Development Agda of February 16, 2014.
```agda
-- M.agda ------------
open import Relation.Binary using (Setoid; module Setoid)
open import Data.Nat        using (ℕ)

h : ℕ
h = 0

module M {α α=} (A : Setoid α α=)
  where
  open Setoid A

  f : Carrier → ℕ
  f _ = 0

  module _ (B : Set)  where
                      g : B → ℕ
                      g _ = 0

-- M2.agda -------------------------------------------
module M2 where

open import Relation.Binary using (Setoid; module Setoid)
open import Data.Nat using (ℕ; suc)
import M

module OverSetoid {α α=} (A : Setoid α α=)
  where
  open Setoid A

  open module MA = M A using (g)

  h : ℕ → ℕ
  h x = suc (g ℕ x)
```

Loading  M2  by  emacs,  C-c C-l reports
```
"The module MA doesn't export the following:  g   ...".  
```
After removing the two lines of   h  from  M.agda, it is type-checked.

Original issue reported on code.google.com by `mech...@botik.ru` on 9 Mar 2014 at 4:14

**Comment by andreasabel** [comment]
We could add a warning if the following conditions are met.

-  Agda reconstructs the top-level module (because of non-import statements like above).
-  The first and only module at the top-level has the same name as the file.

**Comment by andreasabel** [comment]
I implemented this as an error, in fact.

**Comment by andreasabel** [comment]
Maybe the fix was not complete, see #2808.
Agda rejects this:
```agda
foo = Set
module Issue1077 where
```
because the top-level module consists of exactly a single module with the same name as the inferred top-level module.
However, it accepts this:
```agda
foo = Set
module Issue1077 where
bar = Set
```
because now only the top-level module contains as first thing a module that that has the same name as the inferred top-level module.
